OIPA/task_queue/utils.py

```diff
@@ -138,3 +140,62 @@ def await_async_subtasks(started=-1, started_not_set=True):
         check_previous_finished_length = finished
     # After this while loop finishes, we clear the DatasetDownloads tables
     reset_automatic_incremental_parse_dbs()
+
+
+def automatic_incremental_validation(start_at, check_validation):
+    # STEP THREE -- DATASET VALIDATION TASK #
+    # Only execute this step if validation should be active.
+    if start_at in (1, 2, 3) and check_validation:
+        # Prepare checks
+        check_validation_has_started = False
+        check_validation_is_active = False
+        check_empty_iteration_count = 0
+        check_empty_iteration_maximum = 3
+
+        while True:
+            url = "https://iativalidator.iatistandard.org/api/v1/queue/next"
+            response = requests.get(url, timeout=30)
+
+            # If the response is not 200, reset and check back later.
+            if response.status_code != 200:
+                check_validation_has_started = False
+                check_validation_is_active = False
+                time.sleep(60)
+                continue
+
+            check_content_is_empty = response.content.decode("utf-8") == ""
+
+            """
+            Case 1: content empty - started = false - active = false
+                wait for the validator to start
+            Case 2: content has data - started = false - active = false
+                set started to true and active to true
+            Case 3: content has data - started = true - active = true
+                wait for the content to stop having data!
+            Case 4: content empty - started = true - active = true
+                with three iterations, confirm the content is actually empty!
+                set active to false.
+            """
+            # if check_content_is_empty and not check_validation_has_started and not check_validation_is_active:  # NOQA: E501
+            if not check_content_is_empty and not check_validation_has_started and not check_validation_is_active:  # NOQA: E501
+                check_validation_has_started = True
+                check_validation_is_active = True
+            if not check_content_is_empty and check_validation_has_started and check_validation_is_active:  # NOQA: E501
+                check_empty_iteration_count = 0
+            if check_content_is_empty and check_validation_has_started and check_validation_is_active:  # NOQA: E501
+                if check_empty_iteration_count < check_empty_iteration_maximum:
+                    check_empty_iteration_count += 1
+                else:  # Validation has finished
+                    break
```

Comment:
`break` breaks the while loop I guess? Not sure if Python breaks the inner if statement or the while loop, could you verify this?

Reply:
Yeah it breaks the while loop! function tested properly with validation :+1:

Reply:
check!